### PR TITLE
Add system owner [RHCLOUD - 8583]

### PR DESCRIPTION
**Pull request by ryandillinfelton**
Inventory-schemas PR: https://github.com/RedHatInsights/inventory-schemas/pull/30 (to be merged first)

### CODE REVIEW

**Review by thearifismail — APPROVED** [review]
lgtm

**Review by kruai — CHANGES_REQUESTED** [review]
This looks good to me. I think it would also benefit from a test that fails the criteria, though... like create an incorrect owner_id and make sure it doesn't accept it

**Comment by ryandillinfelton** [comment]
> This looks good to me. I think it would also benefit from a test that fails the criteria, though... like create an incorrect owner_id and make sure it doesn't accept it

It wouldn't be possible since we aren't validating any of the system profile right now. I'll make a note to add it when system profile input validation is working.

**Comment by kruai** [comment]
OK, that's fair. In that case, it's good to go!

**Review by kruai — APPROVED** [review]

File: swagger/system_profile.spec.yaml [hunk 1]
```diff
@@ -143,6 +143,9 @@ $defs:
     description: Representation of the system profile fields
     type: object
     properties:
+      owner_id:
```
Comment (jharting) [inline swagger/system_profile.spec.yaml:146]: Changes to system_profile need to go to https://github.com/redhatinsights/inventory-schemas first. The process is documented in https://docs.google.com/document/d/1-2x4awXu3kZCZ5OUAUTxcpcm3IwaZx8D1Q52QVPah1U/edit#

**Review by jharting — CHANGES_REQUESTED** [review]

**Comment by jharting** [comment]
> It wouldn't be possible since we aren't validating any of the system profile right now. I'll make a note to add it when system profile input validation is working.

I don't follow. Inventory has always been validating SP. Why is it not validating SP right now?

**Comment by ryandillinfelton** [comment]
> > It wouldn't be possible since we aren't validating any of the system profile right now. I'll make a note to add it when system profile input validation is working.
> 
> I don't follow. Inventory has always been validating SP. Why is it not validating SP right now?

My bad, I realize now that you say this I was confusing the system profile input with the system profile filters. We still validate the input and I'll add a negative test to check our validation for the field.

**Comment by ryandillinfelton** [comment]
I opened an associated PR on inventory-schemas here: https://github.com/RedHatInsights/inventory-schemas/pull/30

**Review by kruai — CHANGES_REQUESTED** [review]
Re-requesting changes after seeing the above discussion; I just want to make sure the tests are in before I give my final approval

File: swagger/system_profile.spec.yaml [hunk 2]
```diff
@@ -143,6 +143,9 @@ $defs:
...
+        type: string
+        pattern: '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
```
Comment (kruai) [inline swagger/system_profile.spec.yaml:148]: @tahmidefaz brought up a good point on the other PR; can you add an example here too?

Comment (ryandillinfelton) [reply swagger/system_profile.spec.yaml:148]: Yep, just updated it to match all the changes from the other PR

**Review by thearifismail — APPROVED** [review]
LGTM

**Review by kruai — DISMISSED** [review]
LGTM - I just want to make sure @gmcculloug is cool with us calling it "owner_id" for now before merging. It would probably save time overall if we were able to use its final name now

**Comment by kruai** [comment]
@ryandillinfelton It looks like some conflicts need to be resolved before merging first. That might dismiss our approvals again

**Comment by kruai** [comment]
Slack thread on the `owner_id` discussion here: https://ansible.slack.com/archives/CQFKM031T/p1607010514281400
It sounds like we've got the go-ahead at this point, @ryandillinfelton ! I say let's resolve the merge conflicts and get this thing in

**Comment by ryandillinfelton** [comment]
> Slack thread on the `owner_id` discussion here: https://ansible.slack.com/archives/CQFKM031T/p1607010514281400
> It sounds like we've got the go-ahead at this point, @ryandillinfelton ! I say let's resolve the merge conflicts and get this thing in

We need to get https://github.com/RedHatInsights/inventory-schemas/pull/34 merged first. It needs one more approval.

**Review by thearifismail — APPROVED** [review]
looks like `owner_id` is the term @asa is settling on and I agree with it.

**Comment by gmcculloug** [comment]
RedHatInsights/inventory-schemas#34 is now merged.  Let's see if we can address failing checks so this is closer to merging.

**Review by jharting — DISMISSED** [review]

**Review by kruai — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
Looks good.  All tests passed.

**Pull request merged**